tests/blob/test_block_blob.py

```diff
@@ -520,6 +520,55 @@ def test_create_blob_from_path_with_properties(self):
         self.assertEqual(properties.content_settings.content_type, content_settings.content_type)
         self.assertEqual(properties.content_settings.content_language, content_settings.content_language)
 
+    def test_createBlobFromStream_when_short_read_the_whole_stream(self):
+        # to get the actual uploaded blob, can only run live
+        if TestMode.need_recording_file(self.test_mode):
```

Comment:
because we want to make sure if the content uploaded is the same as what we expect to be uploaded. Let's say we want to upload "Hello World" with short read, we want to verify the content uploaded is not "Hello W" or anything else.